### PR TITLE
Add Flag to OAuth2l CLI

### PR DESCRIPTION
**Pull request by LeoRuiz01**

### CODE REVIEW

File: util/tasks.go [hunk 1]
```diff
@@ -96,6 +98,35 @@ func Reset() {
 	}
 }
 
+func Web(url string){
```
Comment (andyrzhao) [inline util/tasks.go:101]: I would put everything web related into a separate file called "web.go" (still under the same util package). Makes it easier to organize when more web-related features and options are added.

File: util/tasks.go [hunk 2]
```diff
@@ -96,6 +98,35 @@ func Reset() {
...
+	cmd := exec.Command("docker-compose", "up -d --build")
+	cmd.Dir = "web"
```
Comment (andyrzhao) [inline util/tasks.go:103]: Can we update this workflow to git clone or install "oauth2l-web" locally if it doesn't exist? OAuth2l will be released as a binary, and all dependencies must be installed separately as a pre-requisite. Installation path of oauth2l-web could be a configurable option for Web.

File: main.go [hunk 3]
```diff
@@ -50,6 +50,7 @@ type commandOptions struct {
 	Info   infoOptions   `command:"info" description:"Display info about an OAuth access token."`
 	Test   infoOptions   `command:"test" description:"Tests an OAuth access token. Returns 0 for valid token."`
 	Reset  resetOptions  `command:"reset" description:"Resets the cache."`
+	Web    webOptions    `command:"web"   description:"Runs the docker images which deploys both frontend and backend for the OAuth2l Playground"`
```
[the remark this answers is not in the document]
Comment (LeoRuiz01) [reply main.go:53]: Fixed!

File: main.go [hunk 4]
```diff
@@ -109,6 +110,10 @@ type resetOptions struct {
 	Cache *string `long:"cache" description:"Path to the credential cache file to remove. Defaults to ~/.oauth2l."`
 }
 
+type webOptions struct {
+	Stop string `long:"stop" description:"Stops the OAuth2l Playground"`
```
[the remark this answers is not in the document]
Comment (LeoRuiz01) [reply main.go:114]: Fixed!

File: main.go [hunk 5]
```diff
@@ -255,6 +269,10 @@ func main() {
 		"test": util.Test,
 	}
 
+	webTasks := map[string](func(string)){
```
[the remark this answers is not in the document]
Comment (LeoRuiz01) [reply main.go:273]: Fixed!

File: util/tasks.go (hunk 1 above)
Comment (LeoRuiz01) [reply util/tasks.go]: Done!

File: main.go [hunk 6]
```diff
@@ -372,8 +390,12 @@ func main() {
 		}
 
 		os.Exit(task(token))
-	} else if cmd == "reset" {
+	} else if task,ok := webTasks[cmd]; ok{
+		url := "http://localhost:3000/"
+		task(url)
+
+	}else if cmd == "reset" {
```
[the remark this answers is not in the document]
Comment (LeoRuiz01) [reply main.go]: Done!

File: main.go [hunk 7]
```diff
@@ -228,6 +233,15 @@ func getInfoOptions(cmdOpts commandOptions, cmd string) infoOptions {
 	}
 	return infoOpts
 }
+//Extracts the web options based on the chosen command
+func getWebOptions(cmdOpts commandOptions, cmd string) webOptions {
```
[the remark this answers is not in the document]
Comment (LeoRuiz01) [reply main.go]: Done!

**Pull request closed without merging**